lib/src/month_view/month_view.dart

```diff
@@ -724,6 +744,32 @@ class _MonthPageBuilder<T> extends StatelessWidget {
       ),
     );
   }
+
+  /// Returns a list of dates for the month, excluding weekends
+  /// if the weekDays parameter is set or showWeekends = false.
+  ///
+  /// This getter first retrieves all the dates of the month using the
+  /// `datesOfMonths` method, considering the `startDay` and
+  /// `hideDaysNotInMonth` parameters.
+  /// It then filters out the weekend dates (Saturday and Sunday)
+  /// if the `weekDays` parameter is set to 5, ensuring only
+  /// weekdays are included in the returned list.
+  List<DateTime> get _filteredDays {
+    final monthDays = date.datesOfMonths(
+      startDay: startDay,
+      hideDaysNotInMonth: hideDaysNotInMonth,
+      showWeekends: weekDays == 7,
+    );
+
+    return monthDays.where((day) {
```

Comment:
can we move this logic in datesOfMonths as we are already passing the flag in the method.

Reply:
Removed this and used `take()` to take only the required dates from the list of week. 7 for Week ends and 5 with hidden weekends.

Reply:
take will cause issue if startDay is otherthan Monday.
![Screenshot 2024-12-19 004237](https://github.com/user-attachments/assets/1006c9c1-076c-4da7-a60e-0efc2b74ee9f)